runtime/doc/builtin.txt

```diff
@@ -3570,6 +3571,17 @@ getloclist({nr} [, {what}])				*getloclist()*
 			:echo getloclist(5, {'filewinid': 0})
 
 
+getmaps()						*getmaps()*
+		Returns a |List| of all mappings. Each List item is a |Dict|
+		as defined by |maparg()|. For a given mapping, the Dict from
+		getmaps() is identical to the dict from |maparg()|.
```

Comment:
A user can use the Dict returned by getmaps() and call mapset() to recreate the mapping?

Reply:
I would think the {mode} argument to mapset() could be a special value and then the "mode" from the dict is uesd.
The new function should also have an "abbr" optional argument to get a list of abbreviations instead of mappings.

Reply:
The test does `assert_equal(getmaps_entry, maparg_return)` for all mappings; how could it not work? (Famous last words)

> have an "abbr" optional argument

And here we go... :-)

So `getmappings([{abbr}])`, if abbr true, then return the abbreviations.

It could be `getmappings({dict})`, where the key `abbr` can be true/false. Just in case there's more options on the way. With that in mind, I'm in favor of just an optional bool for abbreviations.

Reply:
> I would think the {mode} argument to mapset() could be a special value and then the "mode" from the dict is uesd.

I would suggest `mapset({dict})` means take abbr and mode from dict, that's easy for a user to understand. 

It's unclear to me, whether or not `mapcheck()` would want a similar change.

Either way, is there a reason not to include "abbr: true/false" in the mapping-dict? I think it could be done right now, without any negative consequences. I'm guessing `mapset()` wouldn't care today.

I'm a casual vim user, so I'm mostly clueless about a lot of vim stuff.  Considering the above, and playing with the new `maplist()` function, I did the following experiment because I was trying to figure out where `nox` came from for the `Q` mapping. I'm sure whatever drives having `[on]unmap` also implicitly do `sunmap` gives a better user experience.
```
map X xyzzy     [' ']
nunmap X        ['ov']
map X xyzzy     [' ']
xunmap X        ['nos']
map X xyzzy     [' ']
sunmap X        ['nox']
map X xyzzy     [' ']
ounmap X        ['nv']
map X xyzzy     [' ']
sunmap X        ['nox']
smap X xyzzy    ['s', 'nox']
```

Reply:
If overloading functions is philosophically undesirable, `mapset(null, null, dict)` is a possibility.